libraries/AP_Mount/AP_Mount_Siyi.h

```diff
@@ -0,0 +1,192 @@
+        WAITING_FOR_CTRL,
+        WAITING_FOR_DATALEN_LOW,
+        WAITING_FOR_DATALEN_HIGH,
+        WAITING_FOR_SEQ_LOW,
+        WAITING_FOR_SEQ_HIGH,
+        WAITING_FOR_CMDID,
+        WAITING_FOR_DATA,
+        WAITING_FOR_CRC_LOW,
+        WAITING_FOR_CRC_HIGH,
+    };
+
+    // reading incoming packets from gimbal and confirm they are of the correct format
+    // results are held in the _parsed_msg structure
+    void read_incoming_packets();
+
+    // process successfully decoded packets held in the _parsed_msg structure
+    void process_packet();
+
+    // send packet to gimbal
+    void send_packet(SiyiCommandId cmd_id, const uint8_t* databuff, uint8_t databuff_len);
+    void send_1byte_packet(SiyiCommandId cmd_id, uint8_t data_byte);
+
+    // request info from gimbal
+    void request_firmware_version() { send_packet(SiyiCommandId::ACQUIRE_FIRMWARE_VERSION, nullptr, 0); }
+    void request_hardware_id() { send_packet(SiyiCommandId::HARDWARE_ID, nullptr, 0); }
+    void request_configuration() { send_packet(SiyiCommandId::ACQUIRE_GIMBAL_CONFIG_INFO, nullptr, 0); }
+    void request_function_feedback_info() { send_packet(SiyiCommandId::FUNCTION_FEEDBACK_INFO, nullptr, 0); }
+    void request_gimbal_attitude() { send_packet(SiyiCommandId::ACQUIRE_GIMBAL_ATTITUDE, nullptr, 0); }
+
+    // rotate gimbal.  pitch_rate and yaw_rate are scalars in the range -100 ~ +100
+    // yaw_is_ef should be true if gimbal should maintain an earth-frame target (aka lock)
+    void rotate_gimbal(int8_t pitch_scalar, int8_t yaw_scalar, bool yaw_is_ef);
+
+    // center gimbal
+    void center_gimbal();
+
+    // set gimbal's lock vs follow mode
+    // lock should be true if gimbal should maintain an earth-frame target
+    // lock is false to follow / maintain a body-frame target
+    void set_lock(bool lock);
+
+    // send target pitch and yaw rates to gimbal
+    // yaw_is_ef should be true if yaw_rads target is an earth frame rate, false if body_frame
+    void send_target_rates(float pitch_rads, float yaw_rads, bool yaw_is_ef);
+
+    // send target pitch and yaw angles to gimbal
+    // yaw_is_ef should be true if yaw_rad target is an earth frame angle, false if body_frame
+    void send_target_angles(float pitch_rad, float yaw_rad, bool yaw_is_ef);
+
+    // internal variables
+    AP_HAL::UARTDriver *_uart;                      // uart connected to gimbal
+    bool _initialised;                              // true once the driver has been initialised
+    bool _got_firmware_version;                     // true once gimbal firmware version has been received
+
+    // buffer holding bytes from latest packet.  This is only used to calculate the crc
+    uint8_t _msg_buff[AP_MOUNT_SIYI_PACKETLEN_MAX];
+    uint8_t _msg_buff_len;
+    const uint8_t _msg_buff_data_start = 8;         // data starts at this byte of _msg_buff
+
+    // the order of fields matches the incoming message so it can be used to calculate the crc
```

Comment:
Comment doesn't look right - you're not checksumming this structure.  Doesn't need to be packed.  Isn't *really* the "parsed msg" as such, just bits out of it.  More of a "parser_state" if anything....

Reply:
Txs.  Comment changed to, "// parser state and unpacked fields".  The original comment was a hangover from how this driver originally calculated the CRC.  Now this is done from a complete copy of the packet so I've removed the unused ctrl and seq fields from this structure.